lib/3scale_toolbox/commands/proxy_config_command/list_command.rb

```diff
@@ -0,0 +1,81 @@
+module ThreeScaleToolbox
+  module Commands
+    module ProxyConfigCommand
+      module List
+        class ListSubcommand < Cri::CommandRunner
+          include ThreeScaleToolbox::Command
+
+          def self.command
+            Cri::Command.define do
+              name        'list'
+              usage       'list <remote> <service> <environment>'
+              summary     'List Proxy Configurations'
+              description 'List all defined Proxy Configurations'
+              runner ListSubcommand
+
+              param   :remote
+              param   :service_ref
+              param   :environment
+            end
+          end
+
+          def run
+            print_proxy_config_data(proxyconfigs, PROXYCONFIG_FIELDS_TO_SHOW)
+          end
+
+          private
+
+          PROXYCONFIG_FIELDS_TO_SHOW = %w[
+            id version environment
+          ]
+
+          def remote
+            @remote ||= threescale_client(arguments[:remote])
+          end
+
+          def proxy_config_environment
+            arguments[:environment]
+          end
+
+          def service_ref
+            arguments[:service_ref]
+          end
+
+          def proxyconfigs
+            @proxyconfigs ||= list_proxy_configs
+          end
+
+          def list_proxy_configs
+            proxyconfigs = remote.proxy_config_list(service.id, proxy_config_environment)
```

Comment:
Consider having in `Entities::Service.proxy_configs` method and from here call

`service.proxy_configs(proxy_config_environment)`

When requesting a list of children of a service, we can do it using Service entity and return instances of `Entitities::ProxyConfig`

If you do not like this change, at least you should check for errros. All calls to `remote` object should be checked. 

That makes me think, we should wrap remote object and check for errors, so commands or entities do not have to do it. wdyt?

Reply:
I implemented the proxy_configs method.
We can open an issue for the remote redesign.